### PR TITLE
added an additional font

### PR DESCRIPTION
**Pull request by gickowtf**
The font looks similar to the old Nintendo GameBoy font. Furthermore, I have adapted the matrix so that it is possible to display wider and narrower letters so that they do not necessarily have to be three pixels wide.

### CODE REVIEW

File: pixoo/__init__.py [hunk 1]
```diff
@@ -52,7 +53,7 @@ class Channel(IntEnum):
 
 class ImageResampleMode(IntEnum):
     PIXEL_ART = Image.NEAREST
-    SMOOTH = Image.ANTIALIAS
+    #SMOOTH = Image.ANTIALIAS
```
Comment (4ch1m) [inline pixoo/__init__.py]: This is being used in `examples.py`.
Why should this be commented out?

File: pixoo/__init__.py [hunk 2]
```diff
@@ -99,7 +100,7 @@ def __init__(self, address, size=64, debug=False, refresh_connection_automatical
         if self.simulated:
             self.__simulator = Simulator(self, simulation_config)
 
-    def clear(self, rgb=Palette.BLACK):
+    def clear(self, rgb: object = Palette.BLACK) -> object:
```
Comment (4ch1m) [inline pixoo/__init__.py:101]: Why add typing (only) here?

File: pixoo/__init__.py [hunk 3]
```diff
@@ -108,10 +109,11 @@ def clear_rgb(self, r, g, b):
     def draw_character(self, character, xy=(0, 0), rgb=Palette.WHITE):
         matrix = retrieve_glyph(character)
         if matrix is not None:
+            teiler = matrix[-1]
```
Comment (4ch1m) [inline pixoo/__init__.py]: "teiler"?
I guess English is the preferred language for this project.

File: pixoo/__init__.py [hunk 4]
```diff
@@ -245,6 +247,24 @@ def draw_text(self, text, xy=(0, 0), rgb=Palette.WHITE):
...
+                    self.draw_pixel((xy[0] + local_x, xy[1] + local_y), rgb)
+
+    def draw_text2(self, text, xy=(0, 0), rgb=Palette.WHITE):
+        for index, character in enumerate(text):
+            if index == 0:
+                matrix = 0
+            self.draw_character2(character, (matrix + xy[0], xy[1]), rgb)
+            matrix += retrieve_glyph2(character)[-1] + 1
+
+
```
Comment (4ch1m) [inline pixoo/__init__.py]: This is duplicated code.
A more generic approach for `draw_character` and `draw_text` would be better.

File: pixoo/__init__.py [hunk 5]
```diff
@@ -430,6 +450,7 @@ def __send_buffer(self):
             'PicSpeed': 1000,
             'PicData': str(base64.b64encode(bytearray(self.__buffer)).decode())
         }))
+        #print(str(base64.b64encode(bytearray(self.__buffer)).decode()))
```
Comment (4ch1m) [inline pixoo/__init__.py]: Superfluous comment/code.

File: pixoo/__init__.py (hunk 3 above)
Comment (gickowtf) [reply pixoo/__init__.py]: changed to divider, my mistake

File: pixoo/__init__.py (hunk 2 above)
Comment (gickowtf) [reply pixoo/__init__.py:101]: `def clear(self, rgb: object = Palette.BLACK) -> object:`
I don't know why in my main branch it was already like this

File: pixoo/__init__.py (hunk 4 above)
Comment (gickowtf) [reply pixoo/__init__.py]: as also in line 453 I forgot to remove the comments from the programming... i am still quite a beginner but i give my best :D

File: pixoo/__init__.py (hunk 5 above)
Comment (gickowtf) [reply pixoo/__init__.py]: as also in line 453 I forgot to remove the comments from the programming... i am still quite a beginner but i give my best :D

File: pixoo/__init__.py (hunk 1 above)
Comment (gickowtf) [reply pixoo/__init__.py]: as also in line 453 I forgot to remove the comments from the programming... i am still quite a beginner but i give my best :D

File: pixoo/__init__.py (hunk 4 above)
Comment (gickowtf) [reply pixoo/__init__.py]: basically I find the idea very good maybe you can simply select the font by parameter I try it at time to implement.

**Comment by gickowtf** [comment]
@4ch1m  
hey, sorry for all the careless mistakes. i'm quite a beginner but basically already proud that it works at all what i wrote there. i hope you'll forgive me and share my idea with another font.

**Comment by 4ch1m** [comment]
> @4ch1m hey, sorry for all the careless mistakes. i'm quite a beginner but basically already proud that it works at all what i wrote there. i hope you'll forgive me and share my idea with another font.

I'm not the owner/maintainer of this repo. :smile:
I just stumbled on this while checking if there are any updates regarding my PR #20.

File: pixoo/__init__.py (hunk 4 above)
Comment (gickowtf) [reply pixoo/__init__.py]: ok iam Done!
now u can via `from pixoo import Pixoo, SimulatorConfig, FONT_GICKO, FONT_PICO_8 `
include the different Fonts 
in this example u have the my font FONT_GICKO
`pixoo.draw_text('1337', (0, 0), (255, 255, 255), FONT_GICKO)`

**Comment by gickowtf** [comment]
@4ch1m Oh, I see! No problem I have implemented your idea and so is the __init__ only once draw_text() with the new parameter font. I'm currently still working on a font generator with pygame :D

https://github.com/gickowtf/Pixel-Font-Gen

**Comment by SomethingWithComputers** [comment]
Hey I'm a big fan of adding a more fonts! But there's been some major reworking on how the library works. Perhaps you can have another look? I will also look into supporting fonts that are differently sized. However, it'll require some extra data in the way font matrixes are stored currently. Possibly it has to be frontloaded or provided via an extra variable.

https://not-jam.itch.io/not-jam-font-pack

I've recently converted all of these fonts to Bitmap fonts (for use with a Godot pixel art project) and I think some of these will look great on the Divoom as well! So I'll be looking into this in the future :).

**Pull request closed without merging**